cloudsync/tasks.py

```diff
@@ -209,3 +295,21 @@ def parse_content_metadata(response):
         content_length = int(content_length)
 
     return file_name, content_type, content_length
+
+
+def extract_title(filename):
+    """
+    Parses the filename for required course information
+    Args:
+        filename(str): The name of the video file, in format
+        'MIT-<course#>-<year>-<semester>-lec-mit-0000-<recording_date>-<time>-<session>.mp4'
+
+    Returns:
+        A string which should match a collection title: MIT-<course>-<session>-<year>-<semester>
+    """
+    rx = r'([A-Z]+)-([\w\.\s]+)-((\d{4})-(Fall|Spring)-)?lec-mit-0000-(\w+)-(\d+)(-([L\d\-]+))?(_\d+)?.*\.\w'
```

Comment:
I like the `collection_prefix`.
The `collection_suffix` above will be different for every video, as it includes recording date, recording time, and occasionally a room number (for adhoc files) or an iterator for the session ("L01_1", "L01_2").  So we'd end up with a different collection for each file if we used the whole thing.  Need to just extract the session from there if present, and room number for adhoc files (but I'll ask Elaine if room number is really necessary for those).